airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py

```diff
@@ -66,11 +68,14 @@ def get_granted_scopes(self, authenticator):
         try:
             access_token = authenticator.get_access_token()
             url = f"https://api.hubapi.com/oauth/v1/access-tokens/{access_token}"
-            response = requests.get(url=url)
-            response.raise_for_status()
-            response_json = response.json()
-            granted_scopes = response_json["scopes"]
-            return granted_scopes
+            with requests.Session() as s:
```

Comment:
Can we add retry_connection_handler and retry_after_handler from hubspot sources.py as decorators here. This takes care of retryable errors list, logging while retrying etc. This also handles oauth token expired errors we are getting intermittently most likely because of token expiring while fetching scopes. cc: @nidhilashkari17 
(https://app.squadcast.com/incident/63f30acfe8f8a9655487176c)

Reply:
Addressed